h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -814,16 +814,16 @@ private FVecParseWriter makeDout(ParseSetup localSetup, int chunkOff, int nchunk
           localSetup = ParserService.INSTANCE.getByInfo(localSetup._parse_type).setupLocal(vec,localSetup);
           // Zipped file; no parallel decompression;
           InputStream bvs = vec.openStream(_jobKey);
-          ZipInputStream zis = new ZipInputStream(bvs);
+          InputStream dec = decryptionTool.decryptInputStream(bvs);
```

Comment:
good point, they are only closed on a successful competition 

to be fair - closing is really a NO-OP in this scenario, because no resources are really freed by invoking close but it should be cleaned-up, given that it was already there - is it okay to address that in a new PR? @mn-mikke ?

Reply:
Ok, np. Let's keep refactoring changes separated.